lib/shelf-metadata.js

```diff
@@ -0,0 +1,261 @@
+            logger.debug(`Data param: ${data}`);
+
+            if (!property) {
+                throw new Error("Must provide a property to update.");
+            }
+
+            if (!data) {
+                throw new Error("Must provide data to update.");
+            }
+
+            options = this.getOptions(property);
+            options = this.setOptionsData(options, data);
+            logger.info(`Updating metadata for ${options.url}`);
+            logger.debug(`PUT request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.put(options).then((response) => {
+                logger.debug("In request.put().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Creates metadata on the file.
+         *
+         * @param {string} property The metadata property to create.
+         * @param {Object} data The metadata object to create on the file.
+         * @return {Promise.<Object>}
+         */
+        createProperty(property, data) {
+            var options;
+
+            logger.debug("In ShelfMetadata.createProperty()");
+            logger.debug(`Property param: ${property}`);
+            logger.debug(`Data param: ${data}`);
+
+            if (!property) {
+                throw new Error("Must provide a property to create.");
+            }
+
+            if (!data) {
+                throw new Error("Must provide data to create.");
+            }
+
+            options = this.getOptions(property);
+            options = this.setOptionsData(options, data);
+            logger.info(`Creating metadata for ${options.url}`);
+            logger.debug(`POST request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.post(options).then((response) => {
+                logger.debug("In request.post().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Deletes metadata property on the file.
+         *
+         * @param {string} property The metadata property to delete.
+         * @return {Promise.<Object>}
+         */
+        deleteProperty(property) {
+            var options;
+
+            logger.debug("In ShelfMetadata.deleteProperty()");
+            logger.debug(`Property param: ${property}`);
+
+            if (!property) {
+                throw new Error("Must provide a property to delete.");
+            }
+
+            options = this.getOptions(property);
+            logger.info(`Creating metadata for ${options.url}`);
+            logger.debug(`DELETE request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.delete(options)(options).then((response) => {
+                logger.debug("In request.delete().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * @typedef {Object} options
```

Comment:
I suppose it is defined in `request`, but I think it's worthwhile to have the typedef in JSDoc so it's clear to see what's going into it. Would a rename to "requestOptions" or something make sense?